opencti-platform/opencti-front/src/private/components/common/bulk/dialog/BulkRelationDialog.tsx

```diff
@@ -0,0 +1,374 @@
+import { commitMutation, fetchQuery, MESSAGING$ } from 'src/relay/environment';
+import Typography from '@mui/material/Typography';
+import { useFormatter } from 'src/components/i18n';
+import useAuth from 'src/utils/hooks/useAuth';
+import { ArrowRightAlt } from '@mui/icons-material';
+import MenuItem from '@mui/material/MenuItem';
+import { Select, SelectChangeEvent } from '@mui/material';
+import TextField from '@mui/material/TextField';
+import Box from '@mui/material/Box';
+import { StixCoreRelationshipAddInput } from '@components/common/stix_core_relationships/__generated__/StixCoreRelationshipCreationMutation.graphql';
+import { RelayError } from 'src/relay/relayTypes';
+import Loader from 'src/components/Loader';
+import { graphql } from 'react-relay';
+import { allEntitiesKeyList, type StixCoreResultsType } from '../utils/querySearchEntityByText';
+import { getRelationsFromOneEntityToAny, RelationsDataFromEntity, RelationsToEntity } from '../../../../../utils/Relation';
+
+export const searchStixCoreObjectsByRepresentativeQuery = graphql`
+  query BulkRelationDialogQuery(
+    $types: [String]
+    $filters: FilterGroup
+    $search: String
+  ) {
+    stixCoreObjects(types: $types, search: $search, filters: $filters) {
+      edges {
+        node {
+          id
+          entity_type
+          representative {
+            main
+          }
+          objectMarking {
+            id
+            definition_type
+            definition
+            x_opencti_order
+            x_opencti_color
+          }
+          objectLabel {
+            id
+            value
+            color
+          }
+          creators {
+            id
+            name
+          }
+          containersNumber {
+            total
+          }
+        }
+      }
+    }
+  }
+`;
+
+interface BulkRelationDialogProps {
+  stixDomainObjectId: string;
+  stixDomainObjectName: string;
+  stixDomainObjectType: string;
+  isOpen: boolean;
+  onClose: () => void;
+  handleRefetch: () => void;
+}
+
+export interface BulkEntityTypeInfo {
+  representative: string;
+  searchTerm: string;
+  entityName?: string;
+  entityType?: string;
+  index: number;
+  isMatchingEntity: boolean;
+  isExisting: boolean;
+  selectedEntityType: RelationsToEntity;
+  entityTypeList?: entityTypeListType[];
+}
+
+type entityTypeListType = {
+  entity_type: string;
+  id: string;
+};
+
+const classes = {
+  dialog: {
+    '.MuiDialog-paper': {
+      overflowY: 'unset',
+      height: '60vh',
+    },
+  },
+  dialogContent: {
+    '.MuiDialogContent-root': {
+      paddingTop: '20px',
+      overflow: 'initial',
+    },
+  },
+};
+
+const querySearchEntityByText = async (text: string) => {
+  const searchPaginationOptions = {
+    filters: {
+      mode: 'and',
+      filters: [
+        {
+          key: allEntitiesKeyList,
+          values: [text],
+        },
+      ],
+      filterGroups: [],
+    },
+    count: 1,
+  };
+
+  const result = await fetchQuery(
```

Comment:
Usage of fetchQuery should be avoid in new code, we should use more usePreloadedQuery and loadQuery

Reply:
no it's fine for now finally :D